### ISSUE TITLE
install error "bin/ld: cannot find -lz"

### ISSUE THREAD

**Issue by ZhaoL-Bio** [GuoliangLi-HZAU/BatMeth2#14]
Dear Prof. Li,

Thank you so much for your nice tool for DNA methylation analysis.
When I installed BatMeth2, I get an error during "make".

![image](https://user-images.githubusercontent.com/42860482/67455198-837f6100-f65f-11e9-9667-3ff51aaeccb7.png)
I googled it, and got an answer that libz.so cannot be found. However,  libz is in my conda path. I added it to $LD_LIBRARY_PATH, but it still does not work. So, may I get some suggestions from you?
gcc version 4.4.7  gsl-2.4  zlib-1.2.11

Thank you for your time in advance.
Zhao Long

![image](https://user-images.githubusercontent.com/42860482/67458201-263bdd80-f668-11e9-9f98-7cc3634d707a.png)
![image](https://user-images.githubusercontent.com/42860482/67458212-2f2caf00-f668-11e9-85a8-176063eef4a3.png)

**Comment by ZhouQiangwei** [comment]
Dear Zhao Long,

Thank you very much for your interest in the software we developed. 
This error during your installation is due to the GCC that comes with your conda environment. There are two ways to solve this problem.

1. Copy zlib Library in conda environment to GCC Library folder of conda, and then reinstall batmeth2.
cp ~/anaconda2/lib/libz.* ~/anaconda2/x86_64-conda_cos6-linux-gnu/lib/

2. You also can temporarily exit conda environment and install batmeth2.

Please feel free to contact us if you have any questions.

Best Regards,
Qiangwei

**Comment by ZhouQiangwei** [comment]
Dear Zhao Long,

Thank you very much for your interest in the software we developed. 
This error during your installation is due to the GCC that comes with your conda environment. There are two ways to solve this problem.

1. Copy zlib Library in conda environment to GCC Library folder of conda, and then reinstall batmeth2.
cp ~/anaconda2/lib/libz.* ~/anaconda2/x86_64-conda_cos6-linux-gnu/lib/

2. You also can temporarily exit conda environment and install batmeth2.

Please feel free to contact us if you have any questions.

Best Regards,

**Comment by ZhaoL-Bio** [comment]
Hi, Qingwei,
Thank you so much for your reply! Cool! Your solutions worked for me.
Emma... Another error with gsl... 
Is that due to the improper installation of gsl?
![image](https://user-images.githubusercontent.com/42860482/67550068-358a5c00-f738-11e9-93c0-6c842e2f0963.png)

Best,
Zhao Long

**Comment by ZhouQiangwei** [comment]
Hi, Zhao Long,
I am very happy that your problem has been solved.

You can download and install the GSL library and add it to your environment variables.
export C_INCLUDE_PATH=$C_INCLUDE_PATH:~/software/gsl-2.4/include
export CPLUS_INCLUDE_PATH=$CPLUS_INCLUDE_PATH:~/software/gsl-2.4/include
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH::~/gsl-2.4/lib
export LIBRARY_PATH=$LIBRARY_PATH::~/gsl-2.4/lib

**Comment by ZhaoL-Bio** [comment]
Hi, Qiangwei,
Many thanks for your guiding, and installation has been done well.
And this pipeline works well for Arabidopsis, genome size of which is relatively small.
However, there were something looks no logical when I build the index for wheat, which has a big genome.
TAIR10:
![image](https://user-images.githubusercontent.com/42860482/67631060-453aa980-f8cc-11e9-9cdd-96abc6aaa301.png)

Wheat
![image](https://user-images.githubusercontent.com/42860482/67631062-4f5ca800-f8cc-11e9-8e38-6e7a655be380.png)

And I got a message when building index:
![image](https://user-images.githubusercontent.com/42860482/67631070-73b88480-f8cc-11e9-970c-42ef1ca39cc9.png)

So, I just wonder is there a bug for big genome size or I do something wrong.
`BatMeth2 build_index KN9204.fa`
the code for wheat and Arabidopsis are the same

Best Regards
Zhao Long

**Comment by ZhouQiangwei** [comment]
Hi, Zhao long,
Could you please provide a download link for the wheat genome? Thanks.

**Comment by ZhaoL-Bio** [comment]
Hi, Qiangwei,
This is the link for version IWGSC 1.0. https://urgi.versailles.inra.fr/download/iwgsc/IWGSC_RefSeq_Assemblies/v1.0/

Thank you for your time.

Zhao Long

**Comment by ZhouQiangwei** [comment]
Hi, Zhao long,
Thank you very much for your very valuable feedback. We have fixed this bug and hope that you can use the new version of BatMeth2. You need to re-download the fasta genome file before you build the genome index. If you encounter any problems, please feel free to contact us.  I am very sorry for the inconvenience.
Thanks,
Qiangwei

**Comment by ZhaoL-Bio** [comment]
Hi, Qiangwei,
Cool！I will try it and feedback to you. 
Best,
Zhao Long

**Comment by ZhaoL-Bio** [comment]
Hi, Qiangwei,
I got the same error although I downloaded the new version BatMeth2 from github.
I re-installed the new version BatMeth2, and deleted the old index.
But I got the same error.
Then I re-send the genome sequence to the server. And I still got the same error.
I can not re-download the genome, because the genome I want to use is a custom genome assamblied by ourself.
So, are there any suggestions? 
Thank you in advance.

Fortunately, BatMeth2 works for Arabidopsis genome, and I got through the whole process.

Zhao Long
